File: my-folder/0005-longest-palindromic-substring/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if len(s) == 1:
            return s
        
        start_pos = 0
        longest = 0
        for i in range(len(s)):
            l = i
            r = i
            while(l >= 0 and r < len(s) and s[l] == s[r]):
                if(r - l + 1 > longest):
                    start_pos = l
                    longest = r - l + 1 

                l -= 1
                r += 1
            
            l = i
            r = i + 1
            while(l >= 0 and r < len(s) and s[l] == s[r]):
                if(r - l + 1 > longest):
                    start_pos = l
                    longest = r - l + 1
                
                l -= 1
                r += 1
            
        return s[start_pos:start_pos + longest]


                



        

        
        
        

File: my-folder/0005-longest-palindromic-substring/solution.java
class Solution {
    int maxLen = 0;
    int startPtr = 0;
    int endPtr = 0;
    public String longestPalindrome(String s) {
        for(int i = 0; i < s.length(); i++){
            validPalindrome(s, i, i);
            validPalindrome(s, i, i+1);
        }
        return s.substring(startPtr, endPtr + 1);
    }

    private void validPalindrome(String s, int l, int r){
        if(l >= 0 && r < s.length() && s.charAt(l) == s.charAt(r)){
            if(maxLen < r - l + 1){
                maxLen = r - l + 1;
                startPtr = l;
                endPtr = r;
            }
            validPalindrome(s, l - 1, r + 1);
        }   
    }
}
